pkg/parse/parse.go

```diff
@@ -40,6 +41,22 @@ func CommonBuildOptions(c *cli.Context) (*buildah.CommonBuildOptions, error) {
 		memorySwap  int64
 		err         error
 	)
+	nofile := "nofile=1048576:1048576"
+	nproc := "nproc=1048576:1048576"
+
+	if os.Getuid() != 0 || os.Getenv("__rootless__") == "1" {
+		var rlimit unix.Rlimit
+		if err := unix.Getrlimit(unix.RLIMIT_NOFILE, &rlimit); err != nil {
+			return nil, err
+		}
+		nofile = fmt.Sprintf("nofile=%d:%d", rlimit.Cur, rlimit.Max)
+		if err := unix.Getrlimit(unix.RLIMIT_NPROC, &rlimit); err != nil {
+			return nil, err
+		}
+		nproc = fmt.Sprintf("nproc=%d:%d", rlimit.Cur, rlimit.Max)
+	}
+	defaultLimits := []string{nofile, nproc}
+
```

Comment:
I think it'd be less code to initialize `defaultLimits` with `nofile` and `nproc`'s values before checking our privilege level, and just assign `nil` to it if it looks like setting those values would fail.